pkg/cache/queue/manager.go

```diff
@@ -509,24 +526,43 @@ func (m *Manager) RequeueWorkload(ctx context.Context, info *workload.Info, reas
 	return added
 }
 
-func (m *Manager) DeleteWorkload(log logr.Logger, w *kueue.Workload) {
+func (m *Manager) DeleteWorkload(log logr.Logger, wlKey workload.Reference) {
 	m.Lock()
 	defer m.Unlock()
-	m.deleteWorkloadFromQueueAndClusterQueue(log, w, queue.KeyFromWorkload(w))
-	m.DeleteSecondPassWithoutLock(w)
+	m.deleteWorkloadFromAssumedQueueAndClusterQueue(&log, wlKey)
+	m.DeleteSecondPassWithoutLock(wlKey)
 }
 
-func (m *Manager) deleteWorkloadFromQueueAndClusterQueue(log logr.Logger, w *kueue.Workload, qKey queue.LocalQueueReference) {
+func (m *Manager) deleteWorkloadFromObsoleteQueues(log *logr.Logger, wlKey workload.Reference, actualQueue queue.LocalQueueReference) {
```

Comment:
Why does it say "queues", plural? AFAICS you'll always delete from 0 or 1 queues here. Is the plural intended to cover the "0" case? :)